server/bin/pbench-server-put-shim-001.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+# This script is used to prepare the tarballs that a version 002 client
+# submits for further processing. It copies the tarballs and their MD5
+# sums to the archive (after checking) and sets the state links, so
+# that the dispatch script will pick them up and get the ball
+# rolling.
+
+import os
+import sys
+import glob
+import tempfile
+
+from logging import Logger
+from pathlib import Path
+from collections import namedtuple
+
+from pbench.agent.results import CopyResultTb
+from pbench.common.exceptions import BadConfig
+from pbench.common.logger import get_pbench_logger
+from pbench.common.utils import md5sum
+from pbench.server import PbenchServerConfig
+from pbench.server.report import Report
+
+
+_NAME_ = "pbench-server-prep-shim-002"
+
+Results = namedtuple("Results", ["nstatus", "ntotal", "ntbs"])
+
+
+def get_receive_dir(config: PbenchServerConfig, logger: Logger) -> Path:
+
+    # we are explicitly handling version-002 data in this shim
+    receive_dir_prefix = config.get("pbench-server", "pbench-receive-dir-prefix")
+    if not receive_dir_prefix:
+        logger.error("Failed: config value pbench-receive-dir-prefix pbench-server")
```

Comment:
Also, `server/lib/config/pbench-put-shim.cfg` doesn't contain this option...which would be great for exercising this code here, but otherwise seems kind of limiting.

Reply:
Aren't we going to use pbench-server.cfg file itself as config file instead of adding another put-shim.cfg file?

Reply:
As I understand it, when we deploy the shim, we will set up an environment variable which points to the config file; we'll put whatever information the shim needs in that file; and, it could be named _anything_, because it's found via the environment variable -- all that is important is that the shim be able to find the info in it.

So, I think the answer to your question is "yes", more or less.  😀 

My follow-up reply here was perhaps unhelpful, given that you're not actually intending to merge `server/lib/config/pbench-put-shim.cfg` into the `main` branch.  It just struck me as odd, since you went to the trouble of including that config file in the review, that it was missing this option.  😕